src/Ext/WarheadType/Body.cpp

```diff
@@ -243,6 +243,44 @@ void WarheadTypeExt::ExtData::LoadFromINIFile(CCINIClass* const pINI)
 		|| this->InflictLocomotor
 		|| this->RemoveInflictedLocomotor
 	);
+
+	char tempBuffer[32];
+
+	for (size_t i = 0; ; i++)
+	{
+		Nullable<Powerup> crateType;
+		_snprintf_s(tempBuffer, sizeof(tempBuffer), "SpawnsCrate%u.Type", i);
+		crateType.Read(exINI, pSection, tempBuffer);
+
+		if (i == 0 && !crateType.isset())
+			crateType.Read(exINI, pSection, "SpawnsCrate.Type");
+
+		if (!crateType.isset())
+			break;
+
+		if (crateType.isset())
```

Comment:
```suggestion
		if (!crateType.isset())
			break;
```
No need to check twice.